### PR TITLE
Fix the todo generation issues 

### PR DESCRIPTION
**Pull request by MaryGao**
fixes first two issues in https://github.com/Azure/autorest.typescript/issues/2377 and https://github.com/Azure/autorest.typescript/issues/2458

- [x] Support only one success responses
- [x] Handle unsupported credential

### CODE REVIEW

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts [hunk 1]
```diff
@@ -189,14 +190,8 @@ export function getDeserializePrivateFunction(
...
-    const validStatus = [
-      ...new Set(
-        operation.responses
-          .flatMap((r) => r.statusCodes)
-          .filter((s) => s !== "default")
-      )
-    ];
-
+    // Directly using the status code of first response
+    const validStatus = response.statusCodes;
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/helpers/operationHelpers.ts:194]: Directly use the first success status codes.

File: packages/typespec-ts/src/utils/credentialUtils.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  authorization: Authentication
+): HttpAuth[] {
+  const authList = [];
+  for (const option of authorization.options) {
+    for (const auth of option.schemes) {
+      switch (auth.type) {
+        case "apiKey":
+          if (auth.in === "cookie" || auth.in === "query") {
+            // just skip un-supported authentication and report a warning
+            reportDiagnostic(program, {
```
Comment (MaryGao) [inline packages/typespec-ts/src/utils/credentialUtils.ts:22]: Skip un-supported authentication and report a warning

Comment (qiaozha) [reply packages/typespec-ts/src/utils/credentialUtils.ts:22]: What exactly does cookie mean? I feel like it's just set a header with the name cookie, similar like custom key credential?

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts (hunk 1 above)
Comment (qiaozha) [reply packages/typespec-ts/src/modular/helpers/operationHelpers.ts:194]: is this because there's is no real case that has multiple responses in typespec?

Comment (MaryGao) [reply packages/typespec-ts/src/modular/helpers/operationHelpers.ts:194]: yes.

File: packages/typespec-ts/src/utils/credentialUtils.ts (hunk 2 above)
Comment (MaryGao) [reply packages/typespec-ts/src/utils/credentialUtils.ts:22]: https://swagger.io/docs/specification/authentication/cookie-authentication/

Simply speaking I don't think cookie auth is just setting a header with customers' input.

Generally browser would automatically maitain the cookie information and this is transparent for end users when trying to communicate with service side. So the server side could save information like session id in Set-Cookie header and when sending any requests these cookies would be sent by Cookie header properly(with scope or some policy) by browser. 

As for node environemnt, if we try to monitor this process we would have a policy which could do something like BearereAuth, 1. go to somewhere to authenticate the users; 2. if successful we will have Set-Cookie header in response. 3. read this info and maintain cookies info in client; 4. append these cookie info to send real requests.

File: packages/typespec-ts/src/modular/buildCodeModel.ts [hunk 3]
```diff
@@ -1706,15 +1707,14 @@ function emitCredentialParam(
   const auth = getAuthentication(context.program, namespace);
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/buildCodeModel.ts:1707]: @joheredi I was wondering if we need a pr to update the credential type first with the idea of your commit: https://github.com/joheredi/autorest.typescript/commit/dfa6948c7ebf939490cb649263c758475a5f0e52#diff-74f5c341faa7623f0c46c5a1c030d16afc796ec149b852c862b68160ea63c448.

Then my fix could directly be based on that change.

File: packages/typespec-ts/src/lib.ts [hunk 4]
```diff
@@ -237,6 +237,12 @@ const libDef = {
       messages: {
         default: paramMessage`Please note the body type of default response for operation - ${"operationName"} is not a model type.`
       }
+    },
+    "un-supported-credential": {
+      severity: "warning",
+      messages: {
+        default: paramMessage`The defined authentication - ${"credentialType"} is not supported yet and it may require customizations.`
```
Comment (joheredi) [inline packages/typespec-ts/src/lib.ts:244]:
```suggestion
        default: paramMessage`Authentication type ${"credentialType"} is not supported.`
```

File: packages/typespec-ts/src/modular/buildCodeModel.ts (hunk 3 above)
Comment (joheredi) [reply packages/typespec-ts/src/modular/buildCodeModel.ts:1707]: Unfortunately we are not ready to use that yet.

Comment (MaryGao) [reply packages/typespec-ts/src/modular/buildCodeModel.ts:1707]: Sure let's have this fix now.

**Review by qiaozha — APPROVED** [review]

**Pull request merged**